### PR TITLE
fix: Prevents generating a new id if it already exists

### PR DESCRIPTION
**Pull request by soluchok**
Signed-off-by: Andrii Soluk <isoluchok@gmail.com>

### CODE REVIEW

File: pkg/didcomm/protocol/didexchange/service.go [hunk 1]
```diff
@@ -530,12 +530,14 @@ func isNoOp(s state) bool {
 	return ok
 }
 
+// nolint: unparam
 func threadID(didCommMsg service.DIDCommMsg) (string, error) {
-	if didCommMsg.Type() == InvitationMsgType || didCommMsg.Type() == oobMsgType {
+	tID, err := didCommMsg.ThreadID()
```
Comment (llorllale) [inline pkg/didcomm/protocol/didexchange/service.go:535]: when is `didCommMsg.ThreadID()` returning an error? Why can't we remove `threadID()`?

Comment (soluchok) [reply pkg/didcomm/protocol/didexchange/service.go:535]: we can remove it, it returns an error when id and thread id is absent.

Comment (llorllale) [reply pkg/didcomm/protocol/didexchange/service.go:535]: Let's please remove it.

**Comment by llorllale** [comment]
@soluchok can you please remove https://github.com/hyperledger/aries-framework-go/blob/master/pkg/didcomm/protocol/didexchange/states.go#L163 as well?

**Pull request merged**